### ISSUE TITLE
Block Grid with Area specific elements becomes unresponsive when the element with area is removed from the options

### ISSUE THREAD

**Issue by HenrietteWalker** [umbraco/Umbraco-CMS#16046]
### Which Umbraco version are you using? (Please write the *exact* version, example: 10.1.0)

13.2.2

### Bug summary

We’re facing problems where certain pages (content nodes) refuse to load in the backoffice after we delete document types that contain other document types (included in areas).

### Specifics

https://umadmin-acctst.nltg.com/umbraco#/content/content/edit/90826

### Steps to reproduce

1. We’re creating two document types called “Test element” and "Test subelement" in the Settings section. Test element contains an area allowing Test subelement.

![image](https://github.com/umbraco/Umbraco-CMS/assets/166802668/4291ac29-7671-47df-a1eb-6b69eacbb981)
![image](https://github.com/umbraco/Umbraco-CMS/assets/166802668/4b65775d-efa0-4c21-b566-7733782605a4)

2. We add Test element on a page (content node) within the Content section: 

![image](https://github.com/umbraco/Umbraco-CMS/assets/166802668/5d7c559c-6636-447f-b007-847bec73f694)

3. If we delete Test subelement, while keeping Test element, on the content node we receive a message about an unsupported element, and can successfully delete it by clicking the waste bin icon:

![image](https://github.com/umbraco/Umbraco-CMS/assets/166802668/c84b9483-ec90-4a06-914a-a179befbd0ea)

6. If we instead delete Test element under Settings:
 
![image](https://github.com/umbraco/Umbraco-CMS/assets/166802668/bc447c04-463f-42a7-bab7-2a528b19cfa2)

but keep Test subelement, the content page fails to load. We can't delete any element or edit the page at all:

![image](https://github.com/umbraco/Umbraco-CMS/assets/166802668/0ebc1b6d-8e75-4a6a-a0ec-c3a8fc0368da)

### Expected result / actual result

We expect to be able to continue editing the content node after deleting the Test element, not only when we delete only the Test subelement.

**Comment by Migaroez** [comment]
Hey @HenrietteWalker, thank you for the bug report.

It took me a while to figure out what exactly your setup was so I have rewritten your reproduction steps at the bottom of this reply to make it easier to test possible solutions.

The interesting bit here is that this only happens when the removed block definition has an area defined. It doesn't matter if it has a limitation on allowed blocks.

### Updated Steps to reproduce
1. Create 2 element document types, they do not need any properties but it shouldn't influence things either, name them `Main Element` and `Area Element` to keep track of things
2. Create a new Data type of type Block Grid (`Umbraco.BlockGrid`) named `Test Grid`
3. Add the `Area Element` element doctype as a possible block to this datatype without any addition configuration
4. Add the `Main Element` element doctype as another possible block to this datatype. This time before submitting, go to the Areas tab, add an area and add the `Area Element` to `Allowed block Types`
5. Setup a document type with a property backed by the datatype created in step 2.
6. Create, and save, a document node of this document type and configure the grid so it has a `Main Element` with a `area element` inside it
7. Edit the `Test Grid` datatype and remove the `Main Element` from the allowed blocks
8. Go back to edit the document node, the block grid editor is now unresponsive.

**Comment by Migaroez** [comment]
It seems like a rendering thing as the backend returns the (invalid) saved blockgrid value as is
```json
"value": {
                                "contentData": [
                                    {
                                        "contentTypeKey": "38ce2168-e2d1-4e8a-83cb-cc907deacf1c",
                                        "udi": "umb://element/f008b99e390244108a1cf56ccc2f28f3"
                                    },
                                    {
                                        "contentTypeKey": "38ce2168-e2d1-4e8a-83cb-cc907deacf1c",
                                        "udi": "umb://element/d53e90a3af714e539711437a598815e3"
                                    }
                                ],
                                "layout": {
                                    "Umbraco.BlockGrid": [
                                        {
                                            "areas": [
                                                {
                                                    "items": [
                                                        {
                                                            "areas": [
                                                                {
                                                                    "items": [
                                                                    ],
                                                                    "key": "c1b2f5a8-ce37-400f-9055-851f46982fca"
                                                                }
                                                            ],
                                                            "columnSpan": 6,
                                                            "contentUdi": "umb://element/d53e90a3af714e539711437a598815e3",
                                                            "rowSpan": 1
                                                        }
                                                    ],
                                                    "key": "c1b2f5a8-ce37-400f-9055-851f46982fca"
                                                }
                                            ],
                                            "columnSpan": 12,
                                            "contentUdi": "umb://element/f008b99e390244108a1cf56ccc2f28f3",
                                            "rowSpan": 1
                                        }
                                    ]
                                },
                                "settingsData": [
                                ]
                            },
```

**Comment by leekelleher** [comment]
Fixed in #16941, for v13.5.0.
